README.md

```diff
@@ -36,24 +30,58 @@ Encode text and decode tokens:
 fmt.Println("Vocab size:", tk.VocabSize())
 // Vocab size: 30522
 fmt.Println(tk.Encode("brown fox jumps over the lazy dog", false))
-// [2829 4419 14523 2058 1996 13971 3899] [brown fox jumps over the lazy dog]
+// &{[2829 4419 14523 2058 1996 13971 3899] [] [] [] [brown fox jumps over the lazy dog] []}
 fmt.Println(tk.Encode("brown fox jumps over the lazy dog", true))
-// [101 2829 4419 14523 2058 1996 13971 3899 102] [[CLS] brown fox jumps over the lazy dog [SEP]]
+// &{[101 2829 4419 14523 2058 1996 13971 3899 102] [] [] [] [[CLS] brown fox jumps over the lazy dog [SEP]] []}
 fmt.Println(tk.Decode([]uint32{2829, 4419, 14523, 2058, 1996, 13971, 3899}, true))
 // brown fox jumps over the lazy dog
 ```
 
+Encode Result Struct:
+```go
+type Offset struct {
+	Start uint32
+	End   uint32
+}
+
+type TokenizerResult struct {
+	TokenIds          []uint32
+	TypeIds           []uint32
+	SpecialTokensMask []uint32
+	AttentionMask     []uint32
+	Tokens            []string
+	Offsets           []Offset
+}
+```
+
 ## Benchmarks
 ```bash
 go test . -bench=. -benchmem -benchtime=10s
 
 goos: darwin
-goarch: arm64
-pkg: github.com/daulet/tokenizers
-BenchmarkEncodeNTimes-10     	  996556	     11851 ns/op	     116 B/op	       6 allocs/op
-BenchmarkEncodeNChars-10      1000000000	     2.446 ns/op	       0 B/op	       0 allocs/op
-BenchmarkDecodeNTimes-10     	 7286056	      1657 ns/op	     112 B/op	       4 allocs/op
-BenchmarkDecodeNTokens-10    	65191378	     211.0 ns/op	       7 B/op	       0 allocs/op
+goarch: amd64
+pkg: github.com/sunhailin-Leo/tokenizers
+cpu: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
+BenchmarkEncodeNTimes-12                  288102             39637 ns/op             440 B/op         14 allocs/op
+BenchmarkEncodeWithOptionNTimes-12        316530             38884 ns/op             552 B/op         16 allocs/op
+BenchmarkDecodeNTimes-12                  714762             17110 ns/op              96 B/op          3 allocs/op
 PASS
-ok  	github.com/daulet/tokenizers	126.681s
+ok      github.com/sunhailin-Leo/tokenizers     38.997s
+```
+
+## Import this package in your projects
+1、Find your project package main
+```go
+package main
+
+//go:generate make -C $GOPATH/pkg/mod/github.com/sunhailin-!leo/tokenizers@v0.6.0 
+// The last version tag must follow go.mod version.
+
+// Your project imports.
+import ""
+
+func main() {
+	// Your project code here
+}
 ```
+2、Before your first run or build, you must run `go generate` in where did you want to execute `go run / build`
```

Comment:
This is a really nice option! Can you please add an example binary where this works (maybe subdirectory of example, e.g. `example/make` and move current example to `example/prebuilt`), I couldn't make this work locally

Reply:
* Maybe there is some problem with your workspace (I think...). 
* You can open a new demo project including `go.mod, example.go`, and run `go generate` before you run `go run` or `go build`.(I test it work~)
* I think I can add a Dockerfile to explain how it works.